src/event_handler.rs

```diff
@@ -464,93 +470,127 @@ async fn resend_pending_transaction(
-                            pending_txn_data.gas_price =
-                                U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
-                                    * pending_txn_data.gas_price
-                                    / 100;
-                            continue;
-                        }
-                        // Just to be on the safer side, though very less likely to occur because the same txn has been sent successfully once
-                        JobsTxnSendError::GasTooHigh | JobsTxnSendError::ContractExecution => break,
-                        _ => {
-                            sleep(Duration::from_millis(200)).await;
-                            continue;
-                        }
+                // Handle retry logic based on the error enum value
+                match parse_send_error(error_string.to_lowercase()) {
+                    JobsTxnSendError::NonceTooLow => {
+                        // Current nonce is already mined and need not be resent now
+                        break;
                     }
-                };
-
-                // Monitor the newly sent pending txn
-                pending_txn_data.txn_hash = pending_txn.tx_hash();
-                break;
-            }
+                    JobsTxnSendError::OutOfGas => {
+                        pending_txn_data.gas_limit = pending_txn_data.gas_limit + GAS_LIMIT_BUFFER;
+                        continue;
+                    }
+                    JobsTxnSendError::GasPriceLow => {
+                        pending_txn_data.gas_price =
+                            U256::from(100 + RESEND_GAS_PRICE_INCREMENT_PERCENT)
+                                * pending_txn_data.gas_price
+                                / 100;
+                        continue;
+                    }
+                    // Just to be on the safer side, though very less likely to occur because the same txn has been sent successfully once
+                    JobsTxnSendError::GasTooHigh | JobsTxnSendError::ContractExecution => break,
+                    _ => {
+                        sleep(Duration::from_millis(200)).await;
+                        continue;
+                    }
+                }
+            };
 
-            if resend_txn == false {
-                break;
-            }
+            // Monitor the newly sent pending txn
+            pending_txn_data.txn_hash = pending_txn.tx_hash();
+            pending_txn_data.last_monitor_instant = Instant::now();
+            pending_txns
+                .lock()
+                .unwrap()
+                .push_front(pending_txn_data.clone());
+            break;
         }
 
-        // Proceed to the next pending txn in the queue if the current nonce has been resolved or mined
-        if resend_txn == false {
+        // Continue in case the replacement txn was sent successfully or the original pending txn was mined
+        if Instant::now() < pending_txn_data.txn_data.retry_deadline {
```

Comment:
this condition can be true because of break at L574, in rare case if it occurs then the corresponding dummy transaction for nonce would be missed.

Reply:
Yes it can happen, will add check for that as well. Thanks!